packages/blockchain-extension/extension/debug/FabricDebugConfigurationProvider.ts

```diff
@@ -86,30 +86,39 @@ export abstract class FabricDebugConfigurationProvider implements vscode.DebugCo
                 return;
             }
 
-            const runtimes: LocalEnvironment[] = LocalEnvironmentManager.instance().getAllRuntimes();
-
             // At the moment, we only want the user to debug with 1-org environments.
             // See https://github.com/IBM-Blockchain/blockchain-vscode-extension/issues/1995 for the issue of supporting 2-org environments.
             const oneOrgRuntimes: string[] = [];
-            for (const _runtime of runtimes) {
-                if (_runtime.numberOfOrgs === 1) {
-                    const runtimeName: string = _runtime.getName();
-                    oneOrgRuntimes.push(runtimeName);
+            const environmentEntries: FabricEnvironmentRegistryEntry[] = await FabricEnvironmentRegistry.instance().getAll(true); // Get only local entries
+
+            const oneOrgEntries: FabricEnvironmentRegistryEntry[] = environmentEntries.filter((entry: FabricEnvironmentRegistryEntry) => {
+                return entry.numberOfOrgs === 1 && entry.environmentType === EnvironmentType.LOCAL_ENVIRONMENT;
+            });
+
+            for (const entry of oneOrgEntries) {
+                const runtime: LocalEnvironment = await LocalEnvironmentManager.instance().ensureRuntime(entry.name, undefined, 1); // Filtered out 2-org environments.
+                const isRunning: boolean = await runtime.isRunning();
+                if (isRunning) {
+                    oneOrgRuntimes.push(entry.name);
                 }
             }
 
             if (oneOrgRuntimes.length === 0) {
-                outputAdapter.log(LogType.ERROR, `No 1-org local environments found.`);
+                outputAdapter.log(LogType.ERROR, `No 1-org local environments found that are started.`);
                 return;
             }
 
             const chosenEnvironment: string = await UserInputUtil.showQuickPick('Select a 1-org environment to debug', oneOrgRuntimes) as string;
+            if (!chosenEnvironment) {
+                return;
+            }
+            FabricDebugConfigurationProvider.environmentName = chosenEnvironment;
 
             this.runtime = LocalEnvironmentManager.instance().getRuntime(chosenEnvironment);
 
-            const isRunning: boolean = await this.runtime.isRunning();
+            const isChosenRunning: boolean = await this.runtime.isRunning();
```

Comment:
surely you know its already running as you only gave them the choice if running ones?